discovery-provider/alembic/trigger_sql/ddl.sql

```diff
@@ -252,3 +252,41 @@ BEGIN;
 
   delete from users where txhash='0x7bc1e2c100a04061098db053957072b7eb0db75ab8f5c35873b56f6b6b817d9e'; -- tx without handle from another wallet
 COMMIT;
+
+-- 5/31/23 poll trusted notifier for delisted statuses
+BEGIN;
+
+-- Define enums
+DO $$ BEGIN
+    CREATE TYPE delist_entity AS ENUM ('TRACKS', 'USERS');
+EXCEPTION
+    WHEN duplicate_object THEN null;
+END $$;
+
+DO $$ BEGIN
+    CREATE TYPE delist_user_reason AS ENUM ('STRIKE_THRESHOLD', 'COPYRIGHT_SCHOOL', 'MANUAL');
+EXCEPTION
+    WHEN duplicate_object THEN null;
+END $$;
+
+-- Create table to maintain a cursor to the most recent delist status pulled from the host (the only host is trusted notifier)
+CREATE TABLE IF NOT EXISTS delist_status_cursor (
+		host text,
+		entity delist_entity NOT NULL,
+		created_at timestamp with time zone NOT NULL,
+    PRIMARY KEY(host, entity)
+);
+
+-- Create table to store delist statuses for users
+CREATE TABLE IF NOT EXISTS user_delist_statuses (
+  created_at timestamp with time zone NOT NULL,
+  user_id integer NOT NULL,
+  delisted boolean NOT NULL,
+  reason delist_user_reason NOT NULL,
+  PRIMARY KEY(created_at, user_id, delisted)
```

Comment:
thoughts on these pkeys? (sqlalchemy requires primary keys)

Reply:
are these fields sufficient for the pkeys? ik we can have backfilled rows with the same `created_at` but i assume there wouldn't be multiple rows with the same user/track id and `delisted` value in these backfill batches right?

Reply:
technically it's possible, but I just confirmed  that this hasn't happened on staging or prod by running this query:
```sql
SELECT "createdAt", "delisted", "userId", COUNT(*)
FROM "Users"
GROUP BY "createdAt", "delisted", "userId"
HAVING COUNT(*) > 1;
```
this would've only be possible during a backfill, so we don't have to worry about this happening in the future, either.

also just a heads up for tracks it should use trackId (not trackCid) because this has duplicates:
```sql
SELECT "createdAt", "delisted", "trackCid", COUNT(*)
FROM "Tracks"
GROUP BY "createdAt", "delisted", "trackCid"
HAVING COUNT(*) > 1;
```

while this doesn't have any duplicates:
```sql
SELECT "createdAt", "delisted", "trackId", COUNT(*)
FROM "Tracks"
GROUP BY "createdAt", "delisted", "trackId"
HAVING COUNT(*) > 1;
```

Reply:
awesome thanks. i did specify track id rather than cid in the PK :)